src/ProjectTask.php

```diff
@@ -1802,6 +1808,30 @@ public static function recalculatePercentDone($ID)
         return true;
     }
 
+    /**
+     * Recalculate the status of a project task based on the percent_done.
+     * @since 11.0.0
+     * @param array $input
+     */
+    public static function recalculateStatus(array $input)
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        if (!$input['auto_projectstates']) {
+            return false;
+        }
+        $config = \Config::getConfigurationValues('core');
+        if ($input['percent_done'] === 0) {
```

Comment:
```suggestion
        if (!isset($input['auto_projectstates'], $input['percent_done'])) {
            return false;
        }

        if (!$input['auto_projectstates']) {
            return false;
        }
        $config = \Config::getConfigurationValues('core');
        if ($input['percent_done'] === 0) {
```

We must be careful here because some users might try to update a project from the API.
In this case, there is no guarantee that all expected inputs keys will be here.

With that in mind, we should stop the execution if we don't have the needed inputs.